step-api-net/StepApiKeyword/Keyword.cs

```diff
@@ -25,10 +28,28 @@ public class AbstractKeyword
         public TokenSession session;
 
         public TokenSession tokenSession;
-        
+
+        /**
+         * @param e
+         * @return true if the exception passed as argument has to be rethrown.
+         */
         public virtual bool OnError(Exception e)
         {
             return true;
         }
+
+        /**
+         * Hook called before each keyword call.
+         *
+         * @param keyword: the keyword to be called
+         */
+        public virtual void BeforeKeyword(MethodInfo keyword) { }
```

Comment:
@jeromebrongniart Same comment as for Java. Any particular reason for passing the MethodInfo object instead of the annotation instance directly?

Reply:
Same

Reply:
Changed in https://github.com/exense/step-api/pull/18/commits/c878e4ee0690404a68dc2410a2e4926e2c263620